From edc11f89af8abaa6ffe89694bb1d7e7d15316c69 Mon Sep 17 00:00:00 2001
From: Liu Hao <lh_mouse@126.com>
Date: Thu, 26 Apr 2018 01:32:32 +0800
Subject: [PATCH] Make GDB compile on MinGW targets again.

The GNU lib defines `open` and `close` as macros, which interfere
with member functions in the C++ standard library with the same names.
We have to `#undef` them before involving these C++ headers.
---
 gdb/unittests/string_view-selftests.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/gdb/unittests/string_view-selftests.c b/gdb/unittests/string_view-selftests.c
index 182a5df9e4..7cdc2f9213 100644
--- a/gdb/unittests/string_view-selftests.c
+++ b/gdb/unittests/string_view-selftests.c
@@ -25,6 +25,9 @@
 #include "selftest.h"
 #include "common/gdb_string_view.h"
 
+#undef open
+#undef close
+
 /* Used by the included .cc files below.  Included here because the
    included test files are wrapped in a namespace.  */
 #include <string>
-- 
2.17.0

